ctx.go

```diff
@@ -616,3 +616,35 @@ func (c *Ctx) SessSetCacheSize(t int) int {
 func (c *Ctx) SessGetCacheSize() int {
 	return int(C.X_SSL_CTX_sess_get_cache_size(c.ctx))
 }
+
+// DaneEnable initializes shared state required for DANE support. Must be
+// called before any other Dane function.
+// https://www.openssl.org/docs/man1.1.1/man3/SSL_dane_clear_flags.html
+func (c *Ctx) DaneEnable() error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	if C.SSL_CTX_dane_enable(c.ctx) <= 0 {
+		return errorFromErrorQueue()
+	}
+
+	return nil
+}
+
+type DaneFlags int
+
+const (
+	DaneFlagNoDaneEeNamechecks DaneFlags = C.DANE_FLAG_NO_DANE_EE_NAMECHECKS
+)
+
+// DaneSetFlags enables the default flags of every connection associated
+// this context. See DaneFlag for available flags.
+// https://www.openssl.org/docs/man1.1.1/man3/SSL_dane_clear_flags.html
+func (c *Ctx) DaneSetFlags(flags DaneFlags) (oldFlags DaneFlags) {
```

Comment:
```suggestion
func (c *Ctx) DaneSetFlags(flags DaneFlags) DaneFlags {
```
It is not a good practice to use named return parameters to explain what the function returns (plus no other function in this file uses named returns). You can explain the return value in comments above.

Reply:
sure but can you please explain the rationale behind this

Reply:
This is a debatable question in general. For example, you can see https://google.github.io/styleguide/go/decisions.html#named-result-parameters

In your case, named returns were used only for documentation purposes. While it could be done in some rare cases, I believe that documentation should be fully displayed in the comments above the function. You can consider this as a nit tho.